shell/platform/tizen/tizen_view_elementary.cc

```diff
@@ -0,0 +1,374 @@
+            if (self->input_method_context_->IsInputPanelShown()) {
+              handled =
+                  self->input_method_context_->HandleEvasEventKeyUp(key_event);
+            }
+            if (!handled) {
+              self->view_->OnKey(
+                  key_event->key, key_event->string, key_event->compose,
+                  EvasModifierToEcoreEventModifiers(key_event->modifiers),
+                  key_event->keycode, false);
+            }
+          }
+        }
+      };
+  evas_object_event_callback_add(event_layer_, EVAS_CALLBACK_KEY_UP,
+                                 evas_object_callbacks_[EVAS_CALLBACK_KEY_UP],
+                                 this);
+}
+
+void TizenViewElementary::UnregisterEventHandlers() {
+  evas_object_event_callback_del(
+      image_, EVAS_CALLBACK_MOUSE_DOWN,
+      evas_object_callbacks_[EVAS_CALLBACK_MOUSE_DOWN]);
+  evas_object_event_callback_del(
+      image_, EVAS_CALLBACK_MOUSE_UP,
+      evas_object_callbacks_[EVAS_CALLBACK_MOUSE_UP]);
+  evas_object_event_callback_del(
+      image_, EVAS_CALLBACK_MOUSE_MOVE,
+      evas_object_callbacks_[EVAS_CALLBACK_MOUSE_MOVE]);
+  evas_object_event_callback_del(
+      image_, EVAS_CALLBACK_MOUSE_WHEEL,
+      evas_object_callbacks_[EVAS_CALLBACK_MOUSE_WHEEL]);
+  evas_object_event_callback_del(
+      container_, EVAS_CALLBACK_KEY_DOWN,
+      evas_object_callbacks_[EVAS_CALLBACK_KEY_DOWN]);
+  evas_object_event_callback_del(container_, EVAS_CALLBACK_KEY_UP,
+                                 evas_object_callbacks_[EVAS_CALLBACK_KEY_UP]);
+}
+
+TizenViewBase::Geometry TizenViewElementary::GetRenderTargetGeometry() {
+  Geometry result;
+  evas_object_geometry_get(image_, &result.left, &result.top, &result.width,
+                           &result.height);
+  return result;
+}
+
+void TizenViewElementary::SetRenderTargetGeometry(Geometry geometry) {
+  evas_object_resize(container_, geometry.width, geometry.height);
+  evas_object_size_hint_min_set(container_, geometry.width, geometry.height);
+  evas_object_size_hint_max_set(container_, geometry.width, geometry.height);
+  evas_object_move(container_, geometry.left, geometry.top);
+
+  evas_object_resize(event_layer_, geometry.width, geometry.height);
+  evas_object_size_hint_min_set(event_layer_, geometry.width, geometry.height);
+  evas_object_size_hint_max_set(event_layer_, geometry.width, geometry.height);
+  evas_object_move(event_layer_, geometry.left, geometry.top);
+
+  evas_object_resize(image_, geometry.width, geometry.height);
+  evas_object_size_hint_min_set(image_, geometry.width, geometry.height);
+  evas_object_size_hint_max_set(image_, geometry.width, geometry.height);
+  evas_object_move(image_, geometry.left, geometry.top);
+}
+
+TizenViewBase::Geometry TizenViewElementary::GetScreenGeometry() {
+  Geometry result;
+  evas_object_geometry_get(image_, &result.left, &result.top, &result.width,
+                           &result.height);
+  return result;
```

Comment:
The implementation doesn't actually match the method name (the method returns the image's geometry but not the screen geometry). I think `GetScreenGeometry` can be moved to the `TizenWindow` base class as it's only used within `window_channel.cc`.